scapy/layers/tls/tools.py

```diff
@@ -19,10 +22,9 @@ class TLSPlaintext(Packet):
     name = "TLS Plaintext"
     fields_desc = [ ByteEnumField("type", None, _tls_type),
                     ShortEnumField("version", None, _tls_version),
-                    FieldLenField("len", None, length_of="fragment",
-                                  fmt="!H"),
-                    StrLenField("fragment", "",
-                                length_from = lambda pkt: pkt.length) ]
+                    FieldLenField("len", None, length_of="data", fmt="!H"),
+                    StrLenField("data", "",
```

Comment:
Hi!
I don't really see the use of renaming `fragment` into `data` :/ If it's only because it's fancy, then it will break backward compatibility.
Are there any reasons for that?

Reply:
In general a whole layers/tsl/tools has not been tested/used in the past so some changes
(e.g. p.fragment ->p.data - due to collision with p.fragment() method) has been also performed.

Reply:
FBC is not broken because the tls/tools has not been used in the past at all.

Reply:
In general a whole layers/tsl/tools has not been tested/used in the past so some changes
(e.g. p.fragment ->p.data - due to collision with p.fragment() method) has been also performed.

FBC is not broken because the tls/tools has not been used and tested in the past at all.

Reply:
Fair enough !

Reply:
I don't think that this field should be renamed. I think that the original authors designed tools using this module. Renaming fields might break them. Moreover, he is not currently available to reply to our enquiries.

Reply:
Without rename fragment -> data this module simply does not work/unusable.
Additionally:
FBC is not broken because the tls/tools has not been used and tested in the past **at all**.

Reply:
That only means that this TLSPlaintext was never used to build a raw packet =)

The original author was in my team, and I know that this code is used internally. You should no rename fields. See for example in https://github.com/secdev/scapy/blob/master/doc/notebooks/tls/notebook3_tls_compromised.ipynb (it needs to be fixed to support Python3).

The main issue is that the current AEAD decryption was likely never used. It was working in 2017, but I guess that the last September refactoring broke it.

Setting aside the renaming of `fragment` into `data`, your changes look fine and could be merged quickly.

Reply:
Without renaming the code simply does not work.
Maybe in the past Packet class has no self.fragment() method?
It is impossible to change name/remove this method so... ?
What is a way to resolve a problem without renamming fragment -> another name?

Reply:
Ok. Is there a solution but... it makes that external access for p.fragment will be very horrible: p.__getattr__("fragment")

Reply:
So? What is your decision?

Reply:
Ok. I just trying to solve it ("reconcile water with fire" :)

Reply:
Solved. Please wait for pass of test and some cleaups.